src/common/helper.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Contains common helper methods
+ */
+const _ = require('lodash')
+const config = require('config')
+const fs = require('fs')
+const request = require('superagent')
+const parseString = require("xml2js").parseString
+const prefix = require('superagent-prefix')
+const xml2js = require("xml2js");
+
+const logger = require('./logger')
+const m2mAuth = require('tc-core-library-js').auth.m2m
+
+const m2m = m2mAuth(
+  _.pick(config, ['AUTH0_URL', 'AUTH0_AUDIENCE', 'AUTH0_PROXY_SERVER_URL'])
+)
+
+/**
+ * Function to get M2M token
+ * @returns {Promise}
+ */
+async function getM2Mtoken() {
+  return m2m.getMachineToken(config.AUTH0_CLIENT_ID, config.AUTH0_CLIENT_SECRET)
+}
+
+/**
+ * Function to get challenge details with provided query
+ * @param {Object} queryParams query params for filter
+ * @returns {Object} challenge description
+ */
+async function getChallengeDetails(queryParams) {
+  const token = await getM2Mtoken()
+  logger.info(
+    `fetching challenge detail using query params: ${JSON.stringify(
+      queryParams
+    )}`
+  )
+  const response = await getV5Api(token).get('/challenges').query(queryParams)
+  const content = _.get(response.body, '[0]')
+
+  if (content) {
+    return content
+  }
+
+  return null
+}
+
+/**
+ * Function to fetch all the submissions for a given challenge
+ * @param {string} challengeId challengeId
+ * @returns {array} submissions 
```

Comment:
Let's update @returns {array} submissions to `@return {Promise<array>}` - causes issues with linters reporting consumers of this method as using "unnecessary await"

Reply:
![image](https://user-images.githubusercontent.com/1445965/210199865-eebb88b8-0d22-4e6a-a66b-40d88137ea1d.png)